duckduckgo/macos-browser
# Add a custom Debug VPN configuration for Network Protection debugging

Task/Issue URL: https://app.asana.com/0/1199230911884351/1205891680998522/f
Tech Design URL:
CC: @diegoreymendez 

**Description**:

This PR adds a new build configuration dedicated to NetP debugging. This was done so that the default `DuckDuckGo Privacy Browser` scheme no longer needs to install builds to `Applications/DEBUG`, which occasionally caused some problems for other developers. Now, only O-N dev team members will need to use that logic regularly, and everyone else will get the more straightforward debugging approach back.

**Steps to test this PR**:
1. Clean derived data
2. Remove the `Applications/DEBUG` directory
3. Run the `DuckDuckGo Privacy Browser` scheme - make sure everything works, and that NetP is visible but does NOT launch, you should expect to get some kind of system extension error
4. Run the `DuckDuckGo Privacy Browser (VPN Debugging)` scheme - make sure everything works, and that NetP is visible and runs as you would expect



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
